### PR TITLE
The description parameter of @server should be optional

### PR DESCRIPTION
**Pull request by skywing918**
https://github.com/microsoft/typespec/issues/2229

### CODE REVIEW

**Review by wanlwanl — APPROVED** [review]

File: .chronus/changes/serverDescOption-2024-9-21-17-14-49.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+changeKind: fix
+packages:
+  - "@typespec/http"
+---
+
+The description parameter of @server should be optional.
```
Comment (timotheeguerin) [inline .chronus/changes/serverDescOption-2024-9-21-17-14-49.md:7]:
```suggestion
The description parameter of `@server` is now optional
```

File: packages/http/generated-defs/TypeSpec.Http.ts [hunk 2]
```diff
@@ -215,6 +215,12 @@ export type HeadDecorator = (context: DecoratorContext, target: Operation) => vo
  * @example
  * ```typespec
  * @service
+ * @server("https://example.com")
+ * namespace PetStore;
+ * ```
+ * @example
```
Comment (timotheeguerin) [inline packages/http/generated-defs/TypeSpec.Http.ts:221]:
```suggestion

 * @example With a description
```

Comment (timotheeguerin) [reply packages/http/generated-defs/TypeSpec.Http.ts:221]: separate with blank line and add tittle for the 2nd example

File: packages/http/lib/decorators.tsp [hunk 3]
```diff
@@ -270,6 +270,14 @@ extern dec head(target: Operation);
  *
  * ```typespec
  * @service
+ * @server("https://example.com")
+ * namespace PetStore;
+ * ```
+ *
+ * @example
```
Comment (timotheeguerin) [inline packages/http/lib/decorators.tsp:277]:
```suggestion
 *  @example With a description
```

File: packages/http/src/decorators.ts [hunk 4]
```diff
@@ -419,7 +419,7 @@ const VERB_DECORATORS = [$get, $head, $post, $put, $patch, $delete];
 
 export interface HttpServer {
   url: string;
-  description: string;
+  description: string | undefined;
```
Comment (timotheeguerin) [inline packages/http/src/decorators.ts:422]:
```suggestion
  description?: string;
```

File: packages/http/src/decorators.ts [hunk 5]
```diff
@@ -456,11 +456,10 @@ export const $server: ServerDecorator = (
     servers = [];
     context.program.stateMap(HttpStateKeys.servers).set(target, servers);
   }
-  servers.push({
-    url,
-    description,
-    parameters: parameterMap,
-  });
+  const server: HttpServer = description
```
Comment (timotheeguerin) [inline packages/http/src/decorators.ts:459]: you can still do what you did before, if a property is undefined its the same 

 servers.push({
    url,
    description,
    parameters: parameterMap,
  });

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**